lib/updatehub/updatehub.c

```diff
@@ -383,13 +452,40 @@ static void install_update_cb(void)
 		LOG_ERR("Could not receive data");
 		goto cleanup;
 	}
-
 	if (coap_packet_parse(&response_packet, data, rcvd, NULL, 0) < 0) {
 		LOG_ERR("Invalid data received");
 		ctx.code_status = UPDATEHUB_DOWNLOAD_ERROR;
 		goto cleanup;
 	}
 
+	/* check for the duplicate packet based upon block number if
+	 * duplicate discard the packet otherwise process it and
+	 * adjust the timers
+	 */
+
+	int block_num = get_num_block(&response_packet);
```

Comment:
Please introduce variables at the beginning of the block.

Reply:
done